base/src/types.rs

```diff
@@ -657,7 +648,7 @@ impl<Id> ASTType<Id> {
     /// Returns the inner most application of a type application
     pub fn inner_app(&self) -> &ASTType<Id> {
         match **self {
-            Type::App(ref a, _) => a.inner_app(),
+            Type::Data(ref a, _) => a.inner_app(),
```

Comment:
Inner_app is only used in a single place (in the compiler) I think you can remove the method and the call completely. I think that will also fix the ice: field access error